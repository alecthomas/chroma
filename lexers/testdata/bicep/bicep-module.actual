@minLength(3)
@maxLength(11)
param storagePrefix string

param storageSKU string = 'Standard_LRS'
param location string = resourceGroup().location

var uniqueStorageName = '${storagePrefix}${uniqueString(resourceGroup().id)}'

resource stg 'Microsoft.Storage/storageAccounts@2019-04-01' = {
  name: uniqueStorageName
  location: location
  sku: {
    name: storageSKU
  }
  kind: 'StorageV2'
  properties: {
    supportsHttpsTrafficOnly: true
  }
}

module webModule './webApp.bicep' = {
  'name': 'webDeploy'
  'params': {
    'skuName': 'S1'
    'location': location
  }
}

output storageEndpoint object = stg.properties.primaryEndpoints