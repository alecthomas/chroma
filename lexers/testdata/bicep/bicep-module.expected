[
  {"type":"Punctuation","value":"@"},
  {"type":"NameDecorator","value":"minLength"},
  {"type":"Punctuation","value":"("},
  {"type":"Text","value":"3"},
  {"type":"Punctuation","value":")"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Punctuation","value":"@"},
  {"type":"NameDecorator","value":"maxLength"},
  {"type":"Punctuation","value":"("},
  {"type":"Text","value":"11"},
  {"type":"Punctuation","value":")"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"KeywordDeclaration","value":"param"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"storagePrefix"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameFunction","value":"string"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"param"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"storageSKU"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameFunction","value":"string"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringInterpol","value":"'Standard_LRS'"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"KeywordDeclaration","value":"param"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"location"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameFunction","value":"string"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameFunction","value":"resourceGroup"},
  {"type":"Punctuation","value":"()."},
  {"type":"Text","value":"location"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"var"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"uniqueStorageName"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringInterpol","value":"'${storagePrefix}${uniqueString(resourceGroup().id)}'"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"resource"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"stg"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralString","value":"'Microsoft.Storage/storageAccounts@2019-04-01'"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"NameProperty","value":"name"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"uniqueStorageName"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"NameProperty","value":"location"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"location"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"NameProperty","value":"sku"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"NameProperty","value":"name"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"storageSKU"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"NameProperty","value":"kind"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringInterpol","value":"'StorageV2'"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"NameProperty","value":"properties"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"NameProperty","value":"supportsHttpsTrafficOnly"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"true"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"module"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"webModule"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralString","value":"'./webApp.bicep'"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"NameProperty","value":"'name'"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringInterpol","value":"'webDeploy'"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"NameProperty","value":"'params'"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"NameProperty","value":"'skuName'"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringInterpol","value":"'S1'"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"NameProperty","value":"'location'"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"location"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"output"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"storageEndpoint"},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordType","value":"object"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"stg"},
  {"type":"Punctuation","value":"."},
  {"type":"Text","value":"properties"},
  {"type":"Punctuation","value":"."},
  {"type":"Text","value":"primaryEndpoints"}
]
